CodeMaid/Helpers/FileHeaderHelper.cs

```diff
@@ -0,0 +1,143 @@
+                case CodeLanguage.VisualBasic: return Settings.Default.Cleaning_UpdateFileHeaderVB;
+                case CodeLanguage.XAML: return Settings.Default.Cleaning_UpdateFileHeaderXAML;
+                case CodeLanguage.XML: return Settings.Default.Cleaning_UpdateFileHeaderXML;
+                default: return null;
+            }
+        }
+
+        internal static int GetHeaderLength(CodeLanguage language, string docStart)
+        {
+            switch (language)
+            {
+                case CodeLanguage.CSharp:
+                case CodeLanguage.CPlusPlus:
+                case CodeLanguage.JavaScript:
+                case CodeLanguage.LESS:
+                case CodeLanguage.SCSS:
+                case CodeLanguage.TypeScript:
+                    return GetHeaderLength(docStart, "//") +
+                        GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.HTML:
+                case CodeLanguage.XAML:
+                case CodeLanguage.XML:
+                    return GetHeaderLength(docStart, "<!--", "-->");
+
+                case CodeLanguage.CSS:
+                    return GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.PHP:
+                    return GetHeaderLength(docStart, "//") +
+                        GetHeaderLength(docStart, "#") +
+                        GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.PowerShell:
+                    return GetHeaderLength(docStart, "#") +
+                        GetHeaderLength(docStart, "<#", "#>");
+
+                case CodeLanguage.R:
+                    return GetHeaderLength(docStart, "#");
+
+                case CodeLanguage.FSharp:
+                    return GetHeaderLength(docStart, "//") +
+                        GetHeaderLength(docStart, "(*", "*)");
+
+                case CodeLanguage.VisualBasic:
+                    return GetHeaderLength(docStart, "'");
+
+                case CodeLanguage.JSON:
+                case CodeLanguage.Unknown:
+                default:
+                    return 0;
+            }
+        }
+
+        internal static int GetHeaderLength(string docStart, string commentSyntax)
+        {
+            if (!docStart.TrimStart().StartsWith(commentSyntax))
+            {
+                return 0;
+            }
+
+            var separator = new string[] { Environment.NewLine };
+            var lines = docStart.Split(separator, StringSplitOptions.None);
+            var header = new List<string>();
+
+            // adds starting empty lines
+            foreach (var line in lines)
+            {
+                if (!string.IsNullOrWhiteSpace(line))
+                {
+                    break;
+                }
+
+                header.Add(line);
+            }
+
+            // adds comment lines
+            foreach (var line in lines.ToList().Skip(header.Count))
```

Comment:
Would this logic be simpler if it simply counted any header lines and/or blank lines together?  I could see someone having a header format like:

```
// This is my header line

// There are many header lines but this one is mine
```

Reply:
To me, headers are a single block. I add leading empty lines so this block doesn't have to begin at the start of the file. I was thinking about cases like that:

```
// This is my header block
//
// This is my header block
//
// This is my header block

// This is a comment about the below code that I want to keep
```

So I could change it if you like but this was intentional as I often see line breaks in comment blocks added with simple a "//" like in my example above.  You tell me

Reply:
Ok, that makes sense.  I'm OK with assuming it has to be a contiguous section of lines without line breaks for now.